### DISCUSSION TITLE
Is it possible to generate assets (JS and CSS) without Ruby?

### DISCUSSION THREAD

**Discussion by szymonnowak-st** [shakacode/shakapacker#220] [Q&A] [answered]
Hi!

We're currently still on Webpacker 6.0.0 rc5, and we're looking into upgrading to Shakapacker. We're also looking into using https://github.com/preactjs/compressed-size-action GitHub action to monitor assets size, however, we'd prefer to skip installing Ruby and all gems to speed it up. When using Shakapacker, is it possible generate production assets with webpack without installing Ruby?

**Comment by tomdracz** [comment, marked as answer]
@szymonnowak-st Behind the scenes we're simply running webpack compiler.

Running something like `yarn webpack --config path/to/webpack/config.js` will run the compilation just fine and you don't need any Ruby gems (as long as you're not using any `erb` files).

Mind you that this will run webpack specific compilation and output manifest but it will not do stuff like recording compilation digest etc that we use to check if assets require compilation

**Comment by szymonnowak-st** [comment]
Thanks!
